Issue 19 stream Delete

igkon opened the pull request:
By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

Signed-off-by: Igor <hokletka@gmail.com>

  Add invokePathDelete method. That method seals and deletes stream using
dstPath of operation and returns true if deletion was successful.
  The type of input paramater is PathOperation<? extends PathItem>.
Also add invokePathDelete method call in submit method.

Commit:
Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

Signed-off-by: Igor <hokletka@gmail.com>

  Add invokePathDelete method. That method seals and deletes stream using
dstPath of operation and returns true if deletion was successful.
  The type of input paramater is PathOperation<? extends PathItem>.
Also add invokePathDelete method call in submit method.

akurilov commented on src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java:
```diff
@@ -177,6 +178,23 @@ protected int submit(List<O> ops) throws InterruptRunException, IllegalStateExce
+				pathOp.status(Operation.Status.FAIL_UNKNOWN);
+				return false;
+			}
+		}
```
"else": it would be useful to add the error message to the log like
```
Loggers.ERR.debug("Failed to seal the stream {} in the scope {}", stream, scope);
```

akurilov commented on src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java:
```diff
@@ -177,6 +178,23 @@ protected int submit(List<O> ops) throws InterruptRunException, IllegalStateExce
+			if(streamManager.deleteStream(scopeName, streamName)) {
+				return true;
+			}
+			else {
```
log error message similar to the comment below

akurilov commented:
Please update the Delete section in the README file and add the info about the sealing before the stream deletion

igkon committed:
Change docs

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

Signed-off-by: Igor <hokletka@gmail.com>

Change docs for Path Item Operations - Delete.

Commit:
Add information about failures in Delete method into log

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

Signed-off-by: Igor <hokletka@gmail.com>

Add logging when the StreamManager.seal() or StreamManager.delete() fails.

veronikaKochugova reviewed (commented):
After merge https://github.com/emc-mongoose/mongoose-storage-driver-pravega/pull/5 this PR there were conflicts. Please resolve

Commit:
Merge branch 'master' of https://github.com/emc-mongoose/mongoose-storage-driver-pravega into issue-19-streamDelete

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

Signed-off-by: Igor <hokletka@gmail.com>
